proposals/0759-react-native-frameworks.md

```diff
@@ -0,0 +1,452 @@
+
+## Becoming a React Native Framework
+
+In this section, we clarify what are the **requirements** to become an official React Native Framework. We call this also the **Framework Contract** as it describes what are the benefits of- and the asks for being an official Framework.
+
+### Benefits
+
+Frameworks that are adhering to requirements described below:
+
+- Will be spotlighted in the Getting Started page of the React Native website (our highest traffic page). Specifically, we’ll recommend to users to use one of the Frameworks which satisfies the requirements.
+- Will be mentioned during public comms as one of the recommended frameworks by the React Native core team.
+- The company/entity behind the Framework will automatically be granted a [partner status](https://github.com/facebook/react-native/blob/main/ECOSYSTEM.md#partners).
+- PRs and Issues coming from the Framework developers will receive higher priorities.
+
+### Requirements
+
+Here is a short summary of what are the requirements for React Native Frameworks.
+
+#### DOs of Frameworks
+
+- **Focus on developer & user experience** - We believe Frameworks should aim to provide a top-notch developer experience. Frameworks that fail to provide good documentation or developer workflows are not likely high quality frameworks, and our users will most likely not want to use them. Similarly, we want to promote frameworks that provide a top-notch user experience and do not ignore basic user needs, such as accessibility, performance, and delightful UIs.
+- **Focus on building cohesive experiences** - Users are looking into deploying universal apps across multiple targets. We’re delegating this responsibility to frameworks that can provide better experiences to users and craft cohesive developer experiences between mobile, web, desktop and beyond.
+- **Focus on abstracting complexity away** - Any initiative that simplifies the developer workflow is welcomed in the framework domain.
+- **Be popular** - We’ll only be recommending solutions that are widely known and adopted by the community.
+- **Open source** - Recommended frameworks must be free, open source, projects. We allow for companies to provide paid services on top of the Open Source framework, but these services should not be a requirement to use the framework and should not have vendor lock-in.
+
+#### DON’Ts of Frameworks
+
+- **Do not hard fork react-native** - As a rule of thumb, you should not GitHub-fork react-native and re-publish it. If you’re doing so, it’s probably due to a lack of APIs that React Native should provide. Please reach out to us and attempt to de-fork your framework.
+- **Do not soft fork the core APIs** - We want to make sure developers need to learn only once how to write a native component. Building abstraction and wrappers is accepted, forking is not, as it makes it harder for users to share components/modules between targets.
```

Comment:
In those terms, is it acceptable to remove certain parts of `react-native`? For example, I create a React Native framework that removes some props I consider harmful in the `StatusBar` module, or entirely removes certain APIs / Components, such as dropping `Modal` to provide a better alternative.

Reply:
> In those terms, is it acceptable to remove certain parts of react-native? 

As a principe: nope it's not. The reality is a bit different in the sense that we know there are some core components/API with better community alternatives.

Our goal is to work with Framework/Library developers to understand those gaps and streamline the core as much as possible. This follows the same direction of the [Lean Core initiative](https://github.com/facebook/react-native/issues/23313) in 2019. We want the core API to be streamlined and extensible so you don't need to "remove" anything from react-native.